source/agora/common/Hash.d

```diff
@@ -233,3 +203,62 @@ nothrow @nogc @safe unittest
     str.c2 = 'c';
     assert(hashFull(str) == abc_exp);
 }
+
+/*******************************************************************************
+
+    Hashes multiple arguments together
+
+    Params:
+        T = variadic argument types
+        args = the arguments
+
+    Returns:
+        the hash of all the arguments
+
+*******************************************************************************/
+
+public Hash hashMulti (T...)(T args) nothrow @nogc @trusted
+{
+    Hash hash = void;
+    crypto_generichash_state state;
+    crypto_generichash_init(&state, null, 0, Hash.sizeof);
+    scope HashDg dg = (scope const(ubyte)[] data) @trusted
+        => cast(void)crypto_generichash_update(&state, data.ptr, data.length);
+    static foreach (arg; args)
```

Comment:
```suggestion
    static foreach (ref arg; args)
```

But as I mentioned, `static foreach (idx, ref _; args)` because https://issues.dlang.org/show_bug.cgi?id=16521